docs/google/devices.md

```diff
@@ -24,8 +24,179 @@ Several factors can influence this limit, but this can be estimated at about 40
 of total circuit run-time. Circuits that exceed this limit will return a
 "Program too long" error code.
 
+
+### Moment structure
+
+The hardware will attempt to run your circuit as it exists in cirq to the
+extent possible.  The device will respect the moment structure of your circuit
+and will execute successive moments in a serial fashion.
+
+The only exception is that identity gates (such as `cirq.Z ** 0`) and virtual
+Z gates (see below) will not actually perform any hardware actions.
+If a moment contains only these "virtual" gates, it will disappear.
+
+For example, this circuit will execute the two gates in parallel:
+
+```
+cirq.Circuit(
+  cirq.Moment(cirq.X(cirq.GridQubit(4,4)), cirq.X(cirq.GridQubit(4,5)))
+)
+```
+
+This circuit will execute the two gates in serial:
+
+
+```
+cirq.Circuit(
+  cirq.Moment(cirq.X(cirq.GridQubit(4,4))),
+  cirq.Moment(cirq.X(cirq.GridQubit(4,5)))
+)
+```
+
+Lastly, this circuit will only execute one gate, since the first gate
+is virtual and its moment will disappear:
+
+```
+cirq.Circuit(
+  cirq.Moment(cirq.Z(cirq.GridQubit(4,4))),
+  cirq.Moment(cirq.X(cirq.GridQubit(4,5)))
+)
+```
+
+## Gates supported
+
+The following lists the gates supported by Google devices.
+Please note that gate durations are subject to change as hardware isxi
+updated and modified, so please refer to the
+[device specification](https://cirq.readthedocs.io/en/latest/docs/google/specification.html)
+to get up-to-date information on supported gates and durations for
+specific processors
+
+### One qubit gates
+
+Google devices support arbitrary one-qubit gates of any rotation.
+The full complement of these rotations can be accessed by using the
+`cirq.PhasedXZGate`.  More restrictive one-qubit gates, such as
+the Pauli gates `cirq.X`, `cirq.Y`, `cirq.Z`, as well as the
+gate `cirq.PhasedXPowGate` can also be natively executed.
+
+#### Virtual Z gates
+
+Rotation around the Z axis is generally not a hardware operation.
+Instead, the compilation keeps track of the Z phase rotations,
+commuting them forward through the circuit until a non-commuting
+gate is reached.  This compilation is handled automatically for you.
+
+What this means is that `cirq.Z` and `cirq.ZPowGate` gates will
+have zero duration on the device.  Any moments containing only
+these gates will silently disappear from the circuit.
+
+#### Physical Z gates
+
+While most applications prefer shorter circuit durations and
+virtual Z gates, Google hardware does offer the possibility of
+applying a physical Z gate that performs a hardware operation
+to affect the frequency of the qubit.
+
+This can be done by applying a PhysicalZTag to the Z gate,
+such as in the following example:
+
+```
+cirq.Z(cirq.GridQubit(5, 5)).with_tags(cirq.google.PhysicalZTag())
+```
+
```

Comment:
Are these implemented using two microwave pulses?

Reply:
I'm not totally sure actually.  I think I recall that it was some sort of frequency detuning, but I don't remember exactly what the hardware people told me.  Anyway, we don't have implementation details for other gates, so I think it is okay not including this here.